apps/partners-reference/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  ApplicationUpdate,
+  ApplicantUpdate,
+  Language,
+  IncomePeriod,
+  ApplicationSubmissionType,
+  ApplicationStatus,
+  Address,
+} from "@bloom-housing/core"
+import moment from "moment"
+
+/*
+  Some of fields are optional, not active, so it occurs 'undefined' as value.
+  This function eliminates those fields and parse to a proper format.
+*/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+function getAddress(condition, addressData) {
+  const blankAddress: Address = {
+    id: undefined,
+    createdAt: undefined,
+    updatedAt: undefined,
+    street: "",
+    street2: "",
+    city: "",
+    state: "",
+    zipCode: "",
+  }
+
+  return condition ? addressData : blankAddress
+}
+
+export const formatApplicationData = (data: any, listingId, editMode: boolean) => {
+  const language: Language = data.application.language ? data.application.language : null
+
+  const submissionDate: Date | null = (() => {
+    // rename default (wrong property names)
+    const { birthDay: submissionDay, birthMonth: submissionMonth, birthYear: submissionYear } =
+      data.dateSubmitted || {}
+    const { hours, minutes = 0, seconds = 0, time } = data.timeSubmitted || {}
```

Comment:
Where does timeSubmitted comes from? I remember adding dateSubmitted on the backend DTOs, but not this.

Reply:
I see 
```

        <ViewItem label={t("application.add.timeSubmitted")}>
          <TimeField
            id="timeSubmitted"
            name="timeSubmitted"
            label={t("application.add.timeSubmitted")}
            register={register}
            error={!!errors?.timeSubmitted}
            readerOnly
          />
        </ViewItem>
```
but do we save it somewhere?

Reply:
Also can we add a typing to this structure returned by `TimeField` e.g. time is `am` | `pm` and I think that would be useful for other developers to later edit this code.